eng/pipelines/templates/steps/sync-directory.yml

```diff
@@ -121,24 +128,39 @@ steps:
         -AuthToken "$(azuresdk-github-pat)"
 
 - ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
+  - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+    parameters:
+      SkipDefaultCheckout: true
+      Paths:
+        - ${{ parameters.DirectoryToSync }}
+      Repositories:
+        - Name: $(Build.Repository.Name)
+          Commitish: $(Build.SourceVersion)
+          WorkingDirectory: $(System.DefaultWorkingDirectory)/$(Build.Repository.Name)
+        - ${{ each repo in parameters.Repos }}:
+          - Name: ${{ parameters.PROwner }}/${{ repo }}
+            Commitish: ${{ parameters.BaseBranchName }}
+            WorkingDirectory: $(System.DefaultWorkingDirectory)/${{ repo }}
+
   - ${{ each repo in parameters.Repos }}:
     - pwsh: |
         Set-PsDebug -Trace 1
-        git clone --branch ${{ parameters.BaseBranchName }} https://github.com/azure/${{ repo }}
         $repoPath = "${{ repo }}/${{ parameters.DirectoryToSync }}"
-        if (!(Test-Path $repoPath)) { mkdir $repoPath }
-        Remove-Item -v -r $repoPath
-        Copy-Item -v -r $(Build.SourcesDirectory)/${{ parameters.DirectoryToSync }} $repoPath
+        Remove-Item -v -r -ErrorAction Ignore $repoPath
+        Copy-Item -v -r `
+          $(System.DefaultWorkingDirectory)/$(Build.Repository.Name)/${{ parameters.DirectoryToSync }} `
+          $repoPath
         Get-ChildItem -r $repoPath
       displayName: Copy ${{ parameters.DirectoryToSync }} from azure-sdk-tools to ${{ repo }}
       workingDirectory: $(System.DefaultWorkingDirectory)
 
     - template: /eng/common/pipelines/templates/steps/create-pull-request.yml
       parameters:
         RepoName: ${{ repo }}
-        PRBranchName: sync-${{ parameters.DirectoryToSync }}
+        BaseBranchName: ${{ parameters.BaseBranchName }}
+        PRBranchName: bebroder-sync-${{ parameters.DirectoryToSync }}
```

Comment:
I assume this is just for your testing.

Reply:
Whoops didn't mean to push this to the fork branch.